### PR TITLE
Authorize other run api

### PR DESCRIPTION
**Pull request by gaoning777**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/2735)

### CODE REVIEW

File: backend/src/apiserver/server/run_server.go [hunk 1]
```diff
@@ -72,23 +72,35 @@ func (s *RunServer) ListRuns(ctx context.Context, request *api.ListRunsRequest)
 }
 
 func (s *RunServer) ArchiveRun(ctx context.Context, request *api.ArchiveRunRequest) (*empty.Empty, error) {
-	err := s.resourceManager.ArchiveRun(request.Id)
+	err := CanAccessRun(s.resourceManager, ctx, request.Id)
```
Comment (IronPan) [inline backend/src/apiserver/server/run_server.go]: CanAccessRun seems run specific. could you move to this file instead and mark it private, for readability?

Comment (IronPan) [reply backend/src/apiserver/server/run_server.go]: and if you make this function part of RunServer, you could skip passing in resource manager.

Comment (gaoning777) [reply backend/src/apiserver/server/run_server.go]: make sense.
Done

**Pull request merged**